app/controllers/DetailsUpdatedController.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import play.api.Logging
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.ContactHelper
+import views.html.DetailsUpdatedView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class DetailsUpdatedController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  val controllerComponents: MessagesControllerComponents,
+  view: DetailsUpdatedView
+) extends FrontendBaseController
+    with ContactHelper
+    with I18nSupport
+    with Logging {
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) async {
+    implicit request =>
+      val fiName = getFinancialInstitutionName(request.userAnswers)
+      Future.successful(Ok(view(fiName)))
```

Comment:
Since this just returns a page, wrapping it in a `Future.successful(...)` should no longer be required.